ovirt/resource_ovirt_image_transfer.go

```diff
@@ -131,18 +134,34 @@ func resourceOvirtImageTransferCreate(d *schema.ResourceData, meta interface{})
 	log.Printf("starting a transfer for disk id: %s", diskID)
 
 	// initialize an image transfer
-	transferService, err := UploadToDisk(conn, sourceFile, diskID, alias, uploadSize)
+	transferService, err := UploadToDisk(conn, sourceFile, diskID, alias, uploadSize, correlationID)
 	if err != nil {
 		return err
 	}
 
 	log.Printf("finalizing...")
-	_, err = transferService.Finalize().Send()
+	finalizeRequest := transferService.Finalize()
+	finalizeRequest.Query("correlation_id", correlationID)
+	_, err = finalizeRequest.Send()
 
 	if err != nil {
 		return err
 	}
 
+	for {
+		finished, err := allJobsFinished(conn, correlationID)
+		if err != nil {
+			return fmt.Errorf("Failed to fetch jobs %v", err)
+		}
+
+		if finished {
```

Comment:
This may not happen , sometimes you got stuck jobs that you have to clean manually due some networking issues , IIUC this code may end waiting for such jobs forever, wouldn't it be better to have a configurable value after which we exit the loop ?

Reply:
I don't disagree, but as I understand the execution time  is limited by terraform anyway, no? Otherwise all the existing checks are susceptible to this issue as well, since it is more likely for a disk to remain locked after a failure then for jobs to remain in `STARTING`

Reply:
You are right , saw now that disk operations have 10 min timeout in resource_ovirt_disk.go

Reply:
I see we have retry logic in some of the places, but not everywhere:
```golang
		okStateConf := &resource.StateChangeConf{
			Target:     []string{string(ovirtsdk4.DISKSTATUS_OK)},
			Refresh:    DiskStateRefreshFunc(conn, d.Id()),
			Timeout:    d.Timeout(schema.TimeoutUpdate),
			Delay:      10 * time.Second,
			MinTimeout: 3 * time.Second,
		}
		_, err = okStateConf.WaitForState()
		if err != nil {
			log.Printf("[DEBUG] Failed to wait for Disk (%s) to become OK: %s", d.Id(), err)
			return err
		}
		log.Printf("[DEBUG] Disk (%s) has been successfully updated", d.Id())
```

We also have `resource.Retry`:
```golang
	return resource.Retry(2*time.Minute, func() *resource.RetryError {
		log.Printf("[DEBUG] Now to remove Disk (%s)", d.Id())
		_, e := diskService.Remove().Send()
		if e != nil {
			if _, ok := e.(*ovirtsdk4.NotFoundError); ok {
				log.Printf("[DEBUG] Disk (%s) has been removed", d.Id())
				return nil
			}
			return resource.RetryableError(fmt.Errorf("Error removing Disk (%s): %s", d.Id(), e))
		}
		return resource.RetryableError(fmt.Errorf("Disk (%s) is still being removed", d.Id()))
	})

```

So perhaps I should use one of these instead?

Reply:
That's seems better idea to me , please do

Reply:
Updated, I did not test it yet

Reply:
is that tested now ? please let me know when it is ready for merge